components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/vault/VaultTool.java

```diff
@@ -234,6 +234,8 @@ private static void addSystemProperties() {
         }
 
         String secureVaultRegex = properties.getProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY);
-        System.setProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY, secureVaultRegex);
+        if (secureVaultRegex.isEmpty()) {
```

Comment:
We need to check add a null check instead of an empty check

Reply:
Done.